pyccel/parser/semantic.py

```diff
@@ -718,7 +720,8 @@ def _handle_function_args(self, arguments, **settings):
             if isinstance(a.value, StarredArguments):
                 args.extend([FunctionCallArgument(av) for av in a.value.args_var])
             else:
-                if isinstance(a.value, PyccelArithmeticOperator) and a.value.rank:
+                if isinstance(a.value, PyccelArithmeticOperator) and a.value.rank\
+                    or isinstance(a.value, (NumpyNewArray, CudaNewArray)):
```

Comment:
Please add parentheses to separate the and/or statements for readability

Reply:
Done in [bec781c](https://github.com/pyccel/pyccel/pull/1236/commits/bec781ca41cf5261a9fa7a5010ff9052b8502d62).